### PR TITLE
Window resources refactor

### PR DESCRIPTION
**Pull request by alvarosoppelsa**
This PR solves the issue of fps consumption by WindowResources. This is done by avoiding disk access on every update.
Also made a little change on the UI to split folders (header) and files (3 columns at the bottom).
I suck at UI so any suggestion or comment are more than welcome :)
Thanks!

### CODE REVIEW

**Review by armandojaga — CHANGES_REQUESTED** [review]

File: Source/src/ui/WindowResource.cpp [hunk 1]
```diff
@@ -84,4 +122,57 @@ void Hachiko::WindowResource::LoadAsset(const std::string& path)
     {
         App->resources->LoadAsset(path);
     }
-}
\ No newline at end of file
+}
+
+void Hachiko::WindowResource::CreateMaterialGui(const ImVec2 &button_size)
+{
+    if (current_path != "./assets\\materials")
```
[the remark this answers is not in the document]
Comment (alvarosoppelsa) [reply Source/src/ui/WindowResource.cpp:129]: Did you try if this works? Just to be sure

[the remark this answers is not in the document]
Comment (alvarosoppelsa) [reply Source/src/ui/WindowResource.cpp:129]: It does :)

[the remark this answers is not in the document]
Comment (alvarosoppelsa) [reply Source/src/ui/WindowResource.cpp]: Done! I also change a few things if you can give me feedback would be nice

File: Source/src/ui/WindowResource.cpp [hunk 2]
```diff
@@ -7,73 +7,113 @@
...
+
+            if (items_counter > MAX_FILES_PER_COLUMN)
+            {
+                ImGui::NextColumn();
+                items_counter = 0;
+            }
         }
     }
+
+    ImGui::Columns(1);
```
Comment (armandojaga) [inline Source/src/ui/WindowResource.cpp:116]: Columns is the old API way of organizing content, the new one is BeginTable, see Widgets::DragFloat for an example

**Review by armandojaga — APPROVED** [review]
just an optional change to modernize ImGui

**Review by miquelmiro3 — APPROVED** [review]
The UI is a bit clonky but i don't know how to improve it either xD, we can work on it after. Maybe i miss a refresh button since when you import a new model for example it doesn't appear in the folder inmidiately (maybe updating it after a file is dropped into the engine?) but it the same that going back and opening the folder again so well done.

**Comment by alvarosoppelsa** [comment]
> The UI is a bit clonky but i don't know how to improve it either xD, we can work on it after. Maybe i miss a refresh button since when you import a new model for example it doesn't appear in the folder inmidiately (maybe updating it after a file is dropped into the engine?) but it the same that going back and opening the folder again so well done.

Thanks! What do you mean about clonky? When resizing? I'll work on the update button. Thanks for noticing

**Comment by alvarosoppelsa** [comment]
@miquelmiro3 wanna test now? Added reloading when importing resources, do you think the update button is still needed?

**Pull request merged**